src/vma/proto/route_table_mgr.cpp

```diff
@@ -547,87 +511,49 @@ void route_table_mgr::del_route_event(route_val &netlink_route_val)
-			// example: if route table contains entry for- 1.1.1.1/24
-			//			and a new entry for- 1.1.1.1/32 is added
-			// then route might change for some dst ips ---> set as invalid
-			if((common_prefix == (new_dst_addr & new_dst_mask)) && p_val_from_tbl->get_dst_pref_len() < new_dst_pref_len) {
-				p_val_from_tbl->set_state(false);
-				rt_mgr_logdbg("route_val %p is not valid", p_val_from_tbl);
-				p_val_from_tbl->print_route_val();
-			}
-		}
-	}
-
-	rt_mgr_logdbg("route added: dst '%d.%d.%d.%d', netmask '%d.%d.%d.%d', interface '%s'", NIPQUAD(new_dst_addr), NIPQUAD(new_dst_mask), new_if_name);
-
-	if (g_vlogger_level >= VLOG_FUNC) {
-		print_route_tbl();
+	auto_unlocker lock(m_lock);
+	if (netlink_route_val) {
+		route_val* p_route_val = &m_tab.value[m_tab.entries_num];
+		p_route_val->set_dst_addr(netlink_route_val->get_dst_addr());
+		p_route_val->set_dst_mask(netlink_route_val->get_dst_mask());
+		p_route_val->set_dst_pref_len(netlink_route_val->get_dst_pref_len());
+		p_route_val->set_src_addr(netlink_route_val->get_src_addr());
+		p_route_val->set_gw(netlink_route_val->get_gw_addr());
+		p_route_val->set_protocol(netlink_route_val->get_protocol());
+		p_route_val->set_scope(netlink_route_val->get_scope()); 
+		p_route_val->set_type(netlink_route_val->get_type());
+		p_route_val->set_table_id(netlink_route_val->get_table_id());
+		p_route_val->set_if_name(const_cast<char*> (netlink_route_val->get_if_name()));
+		p_route_val->set_state(true);
+		p_route_val->set_str();
+		p_route_val->print_val();
+		
+		++m_tab.entries_num;
 	}
-	update_invalid_entries();
 }
-#endif
-
-#if _BullseyeCoverage
-    #pragma BullseyeCoverage off
-#endif
-bool inline is_route_event(uint16_t event_type)
-{
-	return (event_type == RTM_NEWROUTE || event_type == RTM_DELROUTE);
-}
-#if _BullseyeCoverage
-    #pragma BullseyeCoverage on
-#endif
 
 void route_table_mgr::notify_cb(event *ev)
 {
-	NOT_IN_USE(ev); //TODO remove
-#if 0
-	route_nl_event *route_netlink_ev = dynamic_cast <route_nl_event*>(ev);
-	const netlink_route_info *netlink_route_info = route_netlink_ev->get_route_info();
-	uint16_t event_type = route_netlink_ev->nl_type;
-
-	if(! is_route_event(event_type))
-		return;
-
 	rt_mgr_logdbg("received route event from netlink");
-	route_val netlink_route_val;
-	create_route_val_from_info(netlink_route_info, netlink_route_val);
 
-	switch(event_type)
-	{
-	case RTM_DELROUTE:
-		del_route_event(netlink_route_val);
-		break;
-	case RTM_NEWROUTE:
-		new_route_event(netlink_route_val);
-		break;
-	}
+	route_nl_event *route_netlink_ev = dynamic_cast <route_nl_event*>(ev);
+
+	netlink_route_info* p_netlink_route_info = route_netlink_ev->get_route_info();
+	if (p_netlink_route_info) {
+		switch(route_netlink_ev->nl_type) {
+#if 0
+		case RTM_DELROUTE:
+			del_route_event(p_netlink_route_info->get_route_val());
+			break;
 #endif
+		case RTM_NEWROUTE:
+			new_route_event(p_netlink_route_info->get_route_val());
+			break;
+		}
+		
+	}
```

Comment:
1. Should there not be an `else` clause with error handling/logging?
2. Style: please remove the empty line above (#555) and the 2 bellow (#557, #558).